tools/codegen/api/dispatcher.py

```diff
@@ -72,20 +72,33 @@ def arguments(func: FunctionSchema) -> Sequence[DispatcherArgument]:
             for la in legacy_dispatcher.arguments(func)
         ]
 
+class ProcessTensoroptions(Enum):
+    # TODO GATHER is only needed for non-c10-full ops, remove later.
+    GATHER = 0
+    SCATTER = 1
+    PASS_THROUGH = 2
```

Comment:
nit: I'm personally more a fan of the `ProcessTensoroptions = Enum('ProcessTensoroptions', ('GATHER', 'SCATTER', 'PASS_THROUGH'))` syntax as the numbers get automatically allocated this way. (Also, did you mean TensorOptions, capital-O?)